pulsar/internal/connection.go

```diff
@@ -286,7 +285,10 @@ func (c *connection) run() {
 			if req == nil {
 				return
 			}
-			c.pendingReqs[req.id] = req
+			// does this request expect a response?
+			if req.id != RequestIDNoResponse {
```

Comment:
Rather than having a special value (for users), I'd prefer for the `..NoWait()` variant to not take a request id since it will never be used.

Reply:
That makes sense. Should we make that change in the current pr or do a new one?

Reply:
New PR is fine